src/OrchardCore.Modules/OrchardCore.Users/Extensions/ControllerExtensions.cs

```diff
@@ -60,7 +60,7 @@ internal static async Task<IUser> RegisterUser(this Controller controller, Regis
 
         var settings = await controller.ControllerContext.HttpContext.RequestServices.GetRequiredService<ISiteService>().GetSettingsAsync<RegistrationSettings>();
 
-        if (settings.UsersCanRegister != UserRegistrationType.NoRegistration)
+        if (settings.AllowSiteRegistration)
```

Comment:
Have to add "or allow external login" or probably better trigger an handler that would check if registration should be enabled or not